lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -354,8 +330,16 @@ def test_clear_tools_test86(tools_configuration, groups, remotes, tools):
 
     out, err, exitcode = pytest.helpers.capture(command_refined)
     assert b"" == out
-    assert any(
-        f'pbench-clear-tools: invalid --group option "{x}"'.encode() in err
-        for x in groups.split(",")
-    )
-    assert any(f'No remote host "{x}"'.encode() in err for x in remotes.split(","))
+    if groups != "default":
+        assert any(
+            f'pbench-clear-tools: invalid --group option "{x}"'.encode() in err
+            for x in groups.split(",")
+        )
+    if remotes:
+        assert any(f'No remote host "{x}"'.encode() in err for x in remotes.split(","))
+    if (
+        tools == "wrong_tool1,wrong_tool2"
+        and "wrong_remote" not in remotes
+        and "wrong_group" not in groups
+    ):
+        assert b"Tools ['wrong_tool1', 'wrong_tool2'] not found in remote" in err
```

Comment:
I think this code should look more like this:
```
    if not groups:
        groups = "default"
    for group in groups.split(","):
        err_msg = f'pbench-clear-tools: invalid --group option "{group}"'
        if "wrong" in group:
            assert err_msg.encode() in err
        else
            assert group.encode() not in err

            for remote in remotes.split(","):
                err_msg = f'No remote host "{remote}"'
                if "wrong" in remote:
                    assert err_msg.encode() in err
                else
                    assert remote.encode() not in err

                    wrong_tools = []
                    for tool in tools.split(","):
                        if "wrong" in tool:
                            wrong_tools.append(tool)
                        else:
                            assert tool.encode() not in err                
                    if wrong_tools:
                        wrong_tools = sorted(wrong_tools)
                        err_msg = f"Tools {wrong_tools} not found in remote"
                        assert err_msg.encode() in err
```

Reply:
As we discussed its bit more complicated, let's say this is the combination 
`groups = 'default,wrong_group', remotes = '', tools = 'pidstat,wrong_tool1'` Then it will try to clear all the remotes from the default group but the default consist of the following remotes:
```
"default": {
            "remote1": ["pidstat"],
            "remote2": ["mpstat"],
            "remote3": ["pidstat", "turbostat"],
        }
```
now for the tools that are specified `'pidstat,wrong_tool1'` on the remote2 the error message will say `['pidstat', 'wrong_tool1'] does not exists` so now the following loop woudn't work.
```
                    for tool in tools.split(","):
                        if "wrong" in tool:
                            wrong_tools.append(tool)
                    if wrong_tools:
                        wrong_tools = sorted(wrong_tools)
                        err_msg = f"Tools {wrong_tools} not found in remote"
                        assert err_msg.encode() in err
```

Reply:
Looks like you came up with substantially the same solution that I did.  🙂